Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -166,6 +172,44 @@ class LoginListViewController: UIViewController {
     }
 }
 
+// MARK: - Favicon Loading
+extension LoginListViewController {
+
+    private func loadFaviconsForLogins(logins: [Login]) {
+        logins.forEach { login in
+            loadFaviconsForLoginFromHistory(login).uponQueue(dispatch_get_main_queue()) { result in
+                self.updateRowForLogin(login, withFavicons: result.successValue ?? [])
+            }
+        }
+    }
+
+    private func loadFaviconsForLoginFromHistory(login: Login) -> Deferred<Maybe<[Favicon]>> {
+        return self.profile.favicons.getFaviconsForURL("\(login.hostname)/") >>== { favicons in
```

Comment:
This'll end up as something like `getFaviconsForURLs`, returning `[String: [String]]`.